treeherder/webapp/api/logslice.py

```diff
@@ -78,7 +84,10 @@ def list(self, request, project, jm):
                     elif i >= end_line:
                         break
 
-                    lines.append({"text": line, "index": i})
+                    if format == 'json':
+                        lines.append({"data": json.loads(line), "index": i})
```

Comment:
I'm wondering whether it makes sense for the caller of the API to specify the format like this? Or more: it's common for an API to allow switching between a json and text response - but normally this is just the _response_ rather than the loading of the data in the first place? Should we add a 'format' field to the job_log_url table? Or should we just hardcode some of the names here (ie build-4h is always format text), or is it fine for people to ask for the wrong format and we should just catch the exception and display an appropriate error message? (Currently the exception is unhandled if someone requests format=json on a buildbot text log). 

Lots of questions & I don't really know the best answer either - let's talk on a call soon. (And tbh we can always tweak this post-landing; so long as we catch the exception).